ElementX/Sources/Screens/RoomScreen/View/TimelineView.swift

```diff
@@ -14,68 +14,146 @@
+import SwiftUIIntrospect
+
+struct TimelineView: View {
+    let viewState: TimelineViewState
     @Environment(\.timelineStyle) private var timelineStyle
-    
-    func makeUIViewController(context: Context) -> TimelineTableViewController {
-        let tableViewController = TimelineTableViewController(coordinator: context.coordinator,
-                                                              timelineStyle: timelineStyle,
-                                                              scrollToBottomButtonVisible: $viewModelContext.scrollToBottomButtonVisible,
-                                                              scrollToBottomPublisher: viewModelContext.viewState.scrollToBottomPublisher)
-        return tableViewController
-    }
-    
-    func updateUIViewController(_ uiViewController: TimelineTableViewController, context: Context) {
-        context.coordinator.update(tableViewController: uiViewController, timelineStyle: timelineStyle)
-    }
-    
-    func makeCoordinator() -> Coordinator {
-        Coordinator(viewModelContext: viewModelContext)
-    }
-    
-    // MARK: - Coordinator
-    
-    @MainActor
-    class Coordinator {
-        let context: RoomScreenViewModel.Context
-        
-        init(viewModelContext: RoomScreenViewModel.Context) {
-            context = viewModelContext
-            
-            if viewModelContext.viewState.itemViewModels.isEmpty {
-                viewModelContext.send(viewAction: .paginateBackwards)
+
+    private let bottomID = "RoomTimelineBottomPinIdentifier"
+    private let topID = "RoomTimelineTopPinIdentifier"
+
+    @State private var scrollViewAdapter = ScrollViewAdapter()
+    @State private var paginateBackwardsPublisher = PassthroughSubject<Void, Never>()
+    @State private var scrollToBottomPublisher = PassthroughSubject<Void, Never>()
+    @State private var scrollToBottomButtonVisible = false
+
+    var body: some View {
+        ScrollViewReader { scrollView in
+            ScrollView {
+                bottomPin
+
+                LazyVStack(spacing: 0) {
+                    ForEach(viewState.itemViewStates.reversed()) { viewState in
+                        RoomTimelineItemView(viewState: viewState)
+                            .frame(maxWidth: .infinity, alignment: .leading)
+                            .padding(timelineStyle.rowInsets)
+                            .scaleEffect(x: 1, y: -1)
+                    }
+                }
+
+                topPin
             }
-        }
-        
-        /// Updates the specified table view's properties from the current view state.
-        func update(tableViewController: TimelineTableViewController, timelineStyle: TimelineStyle) {
-            if tableViewController.timelineStyle != timelineStyle {
-                tableViewController.timelineStyle = timelineStyle
+            .introspect(.scrollView, on: .iOS(.v16)) { uiKitScrollView in
```

Comment:
Seems totally reasonable to match the type name here.
```suggestion
            .introspect(.scrollView, on: .iOS(.v16)) { uiScrollView in
```